packages/web/components/TechnologyForm/MapAndAttachments/index.js

```diff
@@ -96,14 +97,19 @@ const MapAndAttachments = ({ form, data }) => {
 		);
 
 		const response = await upload(formData);
-		if (type === 'img') {
-			const newValue = [...previewedImgFiles, ...response];
-			setPreviewedImgFiles(newValue);
-		}
 
-		if (type === 'pdf') {
-			const newValue = [...previewedPdfFiles, ...response];
-			setPreviewedPdfFiles(newValue);
+		if (response.status === 200) {
+			if (type === 'img') {
+				const newValue = [...previewedImgFiles, ...response.data];
+				setPreviewedImgFiles(newValue);
+			}
+
+			if (type === 'pdf') {
+				const newValue = [...previewedPdfFiles, ...response.data];
+				setPreviewedPdfFiles(newValue);
+			}
+		} else {
+			setErrorUploads(response.data.error.message[0].message);
 		}
```

Comment:
```suggestion
		if (response.status !== 200) {
			return setErrorUploads(response.data.error.message[0].message);
		}
		
		if (type === 'img') {
			const newValue = [...previewedImgFiles, ...response.data];
			setPreviewedImgFiles(newValue);
		}
		
		if (type === 'pdf') {
			const newValue = [...previewedPdfFiles, ...response.data];
			setPreviewedPdfFiles(newValue);
		}
```

Reply:
É justamente nessa parte que aparece o bug, aparece o botão de apagar sem nenhuma imagem. 
Nesse caso mesmo se o status for 400, vai injetar um objeto indefinido no array.

Reply:
Sim sim, foi só uma refatoração no if :slightly_smiling_face: